frontend/src/components/ui/WorkspaceTabs.jsx

```diff
@@ -27,9 +32,22 @@ export default function WorkspaceTabs() {
   const handleTabChange = (evt) => {
     const index = evt.selectedIndex;
     const key = renderedTabs[index]?.id;
+
+    setZoomLevels((prevZoomLevels) => ({
+      ...prevZoomLevels,
+      [workspace.active]: editor?.zoom,
+    }));
+
     setWorkspace((draft) => {
       draft.active = key;
     });
+
+    if (editor) {
+      editor.zoom = zoom_levels[key] !== undefined ? zoom_levels[key] : 1;
```

Comment:
In JS, undefined is actually "Falsy" so and equivalent shorter statement would be 
```js
zoom_levels[key] ? zoom_levels[key] : 1;
``` 
And since you're setting the same value that you're testing, you can make it even shorter with the nullish coalescing operator.
```js
zoom_levels[key] ??  1;
```

https://developer.mozilla.org/en-US/docs/Glossary/Falsy
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

Reply:
Changed it accordingly.